src/main/java/com/Assignment/AssignmentFour/entity/Item.java

```diff
@@ -0,0 +1,149 @@
+package com.Assignment.AssignmentFour.entity;
+
+import org.hibernate.annotations.GenericGenerator;
+import org.springframework.data.annotation.Version;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+@Entity
+/**
+ * Entity class.
+ */
+public class Item {
+  @Id @GeneratedValue(generator="system-uuid")
+  @GenericGenerator(name="system-uuid", strategy = "uuid")
+  /**
+   * Item id.
+   */
+  private String id;
+  /**
+   * Item name.
+   */
+  private String name;
+  /**
+   * Item quantity.
+   */
+  private int quantity;
+  /**
+   * Item price.
+   */
+  private int price;
+  /**
+   * Item type.
+   */
+  private String type;
+  /**
+   * Sales tax calculated.
+   */
+  private double salesTax;
+  /**
+   * Final price of item.
+   */
+  private double finalPrice;
+  @Version
+  /**
+   * Variable to implement optimistic locking.
+   */
+  private int version;
+
+  /**
+   * Constructor for Item class.
+   * @param name name
+   * @param type type
+   * @param price price
+   * @param quantity quantity
+   */
+  public Item(final String name, final String type,final int price,final int quantity){
+    this.name=name;
+    this.price=price;
+    this.quantity=quantity;
+    this.type=type;
+    setFinalPrice();
+  }
+
+  /**
+   * Default constructor.
+   */
+  public Item(){
+
+  }
+
+  /**
+   *Method to get quantity.
+   * @return quantity
+   */
+  public int getQuantity() {
+    return quantity;
+  }
+  /**
+   *Method to get price.
+   * @return price
+   */
+  public int getPrice() {
+    return price;
+  }
+  /**
+   *Method to get type.
+   * @return type
+   */
+  public String getType() {
+    return type;
+  }
+  /**
+   *Method to get salestax.
+   * @return salestax
+   */
+  public double getSalesTax() {
+    return salesTax;
+  }
+  /**
+   *Method to get final price.
+   * @return final price
+   */
+  public double getFinalPrice() {
+    return finalPrice;
+  }
+
+  public void setFinalPrice() {
```

Comment:
abstract this logic out of the entity. Can use builder pattern.

Reply:
Created itembuilder class.